### PR TITLE
Clarify Public ID usage in metadata upload

### PR DESCRIPTION
**Pull request by vincent-czi**
### Summary:
- **What:** Tweaks to metadata upload flow to clarify how user should interact with Public (GISAID) ID. Also some other fixes.
- **Ticket:** [sc<177067>](https://app.shortcut.com/genepi/story/<177067>); [sc<181309>](https://app.shortcut.com/genepi/story/<181309>); [sc<182546>](https://app.shortcut.com/genepi/story/<182546>); [sc<143531>](https://app.shortcut.com/genepi/story/<143531>)
- **Env:** https://public-id-meta-tweaks-frontend.dev.czgenepi.org

### Demos:

Public ID column has moved and been renamed, toggle for "Previously Submitted to GISAID?" is gone
image

-----

Now has the date of last change to metadata template structure
image

-----

Additional explanation to user they should check current metadata template when they have incorrect TSV/CSV headers for metadata file [This will change down the road once Christine's usage instructions are complete, will become a link out to that then.]
image

### Notes:

A sample can now be both private (private within app, not exposed to "parent" orgs of a group) and have a public ID (submitted to GISAID or similar). Previously, we would auto-correct uploaded TSV data so that a sample could not be both of these at once, but we've changed our mind on how that should work. Since this was the only form of auto-correcting that was occurring, I've removed all the warning components and functions doing the auto-correcting. However, there is another flavor of auto-corrrect that is in upcoming product specifications for CRUD, so I left the warning type in the enum. 

### Checklist:
- [x] I merged latest `trunk`
- [x] I manually verified the change
- [x] I added labels to my PR
- [x] I tested in multiple browsers
- [ ] I added relevant unit tests
~[ ] I have notified others of changes they need to make locally (migrations, jobs, package updates, etc)~ (none needed)

### CODE REVIEW

File: src/frontend/src/views/Upload/components/Metadata/components/ImportFile/prepMetadataTemplate.ts [hunk 1]
```diff
@@ -3,17 +3,20 @@
  */
 import { METADATA_KEYS_TO_HEADERS } from "../../../common/constants";
 
+// Should change below whenever there are material changes to TSV download
+export const TEMPLATE_UPDATED_DATE = "2022-02-03"; // YYYY-MM-DD
```
Comment (vincent-czi) [inline src/frontend/src/views/Upload/components/Metadata/components/ImportFile/prepMetadataTemplate.ts]: 2022-02-03 is the likely date of this being deployed to Prod, assuming it gets merged in before then. If it isn't merged before the next Prod push, I'll update this to whatever the next Prod push date will be.

**Review by lvreynoso — APPROVED** [review]
lgtm

**Comment by vincent-czi** [comment]
I've re-requested review because I pushed up a fairly major chunk of new commits since approval.

Commits address mostly address visual design stuff after some conversations with Janeece.

File: src/frontend/src/views/Upload/components/Metadata/components/ImportFile/components/Alerts/common/ProblemTable.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+ * Very simple table that uses zebra-striped rows to display info.
+ *
+ * Displays column headers at top of table with each row beneath.
+ * Assumes that the order of column headers and each row inside of `rows`
+ * is matched already. There is no extraction based on object keys or anything,
+ * it's just a simple table for when we don't need a complex data view.
+ *
+ * Currently just intended for use in Metadata warnings and errors because
+ * all those components -- when showing issues via a table -- share this design
+ * design for the expanded message showing everything going wrong.
```
Comment (phoenixAja) [inline src/frontend/src/views/Upload/components/Metadata/components/ImportFile/components/Alerts/common/ProblemTable.tsx:27]:
```suggestion
 * all those components -- when showing issues via a table -- share this
 * design for the expanded message showing everything going wrong.
```

File: src/frontend/src/views/Upload/components/Metadata/components/ImportFile/prepMetadataTemplate.ts [hunk 3]
```diff
@@ -3,17 +3,20 @@
  */
 import { METADATA_KEYS_TO_HEADERS } from "../../../common/constants";
 
+// Should change below whenever there are material changes to TSV download
+export const TEMPLATE_UPDATED_DATE = "2022-02-08"; // YYYY-MM-DD
```
Comment (phoenixAja) [inline src/frontend/src/views/Upload/components/Metadata/components/ImportFile/prepMetadataTemplate.ts:7]:
```suggestion
export const TEMPLATE_UPDATED_DATE = "2022-02-09"; // YYYY-MM-DD
```

**Review by phoenixAja — APPROVED** [review]

**Comment by vincent-czi** [comment]
Switching over to doing this work as a long-lived feature branch. Should not merge this branch in, but will instead batch up all the Upload flow changes into a single branch (with smaller PRs made against that branch to keep ourselves sane), at which point it can all be merged at once.

Long-lived PR / branch here: https://github.com/chanzuckerberg/czgenepi/pull/944

File: src/frontend/src/views/Upload/components/Metadata/components/ImportFile/prepMetadataTemplate.ts (hunk 3 above)
Comment (vincent-czi) [reply src/frontend/src/views/Upload/components/Metadata/components/ImportFile/prepMetadataTemplate.ts:7]: I'll bump this date corresponding to when the big feature branch will get merged, but thanks for the reminder! I'll make a note to myself to take care of that at the end.

**Pull request closed without merging**